datasets/dataset_cifar10.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""
+CIFAR10 dataset.
+"""
+
+from keras.datasets import cifar10
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Activation, Flatten
+from keras.layers import Conv2D, MaxPooling2D
+import keras
+
+
+# Data samples pre-processing method for inputs
+def preprocess_dataset_inputs(x):
+
+    x = x.astype("float32")
+    x /= 255
+
+    return x
+
+
+# Data samples pre-processing method for labels
+def preprocess_dataset_labels(y):
+
+    y = keras.utils.to_categorical(y, num_classes)
+
+    return y
+
+
+# Load and flatten data
+(x_train, y_train), (x_test, y_test) = cifar10.load_data()
+y_train = y_train.flatten()
+y_test = y_test.flatten()
+
+# Init dataset-specific variables
+input_shape = (32, 32, 3)
+num_classes = 10
+
+# Pre-process inputs
+x_train = preprocess_dataset_inputs(x_train)
+x_test = preprocess_dataset_inputs(x_test)
```

Comment:
I don't understand where is this used ?

Reply:
Just here: inputs are pre-processed on the spot here.

Reply:
Ok I understand now !